torchqrnn/_cpu_forget_mult.pyx

```diff
@@ -0,0 +1,50 @@
+# cython: infer_types=True
+#
+
+cdef void recurrent_forget_mult(float *dst, const float* F, const float* X,
+        int nN, int nB, int nO) nogil:
+    cdef int ts, b, h, prev_step, this_step
+    # ts is timestep index, b is batch index, h is hidden index
+    cdef int i = 0
+    for ts in range(1, nN+1):
+        # To move timesteps, we step HIDDEN * BATCH
+        # To move batches, we move HIDDEN
+        # To move neurons, we move +- 1
+        prev_step = (ts-1) * nO * nB
+        this_step = (ts-0) * nO * nB
+        for b in range(nB):
+            prev_step += b * nO
```

Comment:
Won't this "overshoot" as we're updating `prev_step`/`this_step` in both the `range(nO)` loop and the `range(nB)` loop?

Reply:
If we had fancy indexing we'd want this_step = (ts, b, h) and prev_step = (ts-1, b, h) right?

So we need this_step = ts * timestamp_stride + b * batch_stride + h * h_stride

My intention was to set to ts * timestamp_stride in the outer loop, and add in the batch stride in the middle loop. The inner calculation is so simple it seemed we would be doing more work on index calculation than actual computation.